js/course_settings_add_school_detail.js

```diff
@@ -27,32 +27,65 @@ function fetchAccountParents(account) {
 }
 
 function handleResponse(response) {
-    if(response.ok) {
-        return response.json();
-    }
-    throw new Error('API request ' + response.url + ' failed with status code ' + response.status);
+    if (response.ok) return response.json();
+
+    // if the request is unauthorized for an account API call, try to provide a useful fallback
+    const accountInfo = getFallbackAccountInfo(response.url);
+    if (accountInfo !== null) return accountInfo;
+
+    // this is either not an account fetch error, or it's an unhandled account fetch error,
+    // so raise it as we're not expecting it
+    const msg = 'API request ' + response.url + ' failed with status code ' + response.status;
+    throw new Error(msg);
+}
+
+function getFallbackAccountInfo(url) {
+    const reAccountUrl = /\/api\/v1\/accounts\/(?<accountId>\d+)$/;
+    const match = reAccountUrl.exec(url);
+    const accountId = match?.groups?.accountId;
+
+    if (!accountId) return null; // this is not an account API call
+
+    /* This user does not have access to the account `accountId`,
+        and will likely be unable to fetch any parent accounts,
+        so simply show the root account `name`; without an `id`
+        this node should be rendered as text, not a link, and
+        without a `parent_account_id` there will be no more
+        account API fetches after this one.
+    */
+    return {name: 'Harvard University'};
 }
 
 function getAccountsPath(accounts) {
     return accounts.slice().reverse().map(function(account) {
-        return '<a href="/accounts/'+account.id+'">'+account.name+'</a>';
+        return account.id ? '<a href="/accounts/'+account.id+'">'+account.name+'</a>' : account.name;
     }).join(" > ");
 }
 
 function annotatePage(text, style) {
     style = style || {};
     style.color = style.color || "#000";
     style.backgroundColor = style.backgroundColor || "#ddd";
-    var span, el = document.getElementById("course_account_id");
```

Comment:
Note this is changed to a `<div>` because `<div>` elements can have top margins next to `<span>` elements, while two sibling `<span>` elements do not render the margin.